assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -468,18 +468,24 @@ const baseActions = {
 	 * @param {string} measurementID Measurement ID.
 	 */
 	*updateSettingsForMeasurementID( measurementID ) {
-		const registry = yield commonActions.getRegistry();
+		const { select, dispatch, __experimentalResolveSelect } =
+			yield commonActions.getRegistry();
 
-		registry
-			.dispatch( MODULES_ANALYTICS_4 )
-			.setMeasurementID( measurementID );
+		dispatch( MODULES_ANALYTICS_4 ).setMeasurementID( measurementID );
 
 		if ( ! isFeatureEnabled( 'gteSupport' ) ) {
 			return;
 		}
+		// Wait for authentication to be resolved to check scopes.
+		yield commonActions.await(
+			__experimentalResolveSelect( CORE_USER ).getAuthentication()
+		);
+		if ( ! select( CORE_USER ).hasScope( TAGMANAGER_READ_SCOPE ) ) {
+			return;
+		}
```

Comment:
Discovered during testing, without this guard, it is possible to see a non-fatal error in the GA interface during setup if `gteSupport` is enabled but the scope has not been granted yet, such as in this scenario. This condition should generally checked almost everywhere we use the `gteSupport` feature flag.